client/app/classView/commentListWidget/commentListWidget.controller.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+angular.module('classViewApp')
+	.controller('CommentListWidgetCtrl', ($scope, $q, _, Comment, moment) => {
+    // update list of comments when commentWidget makes a new comment
+    $scope.$on('commentListNewComment', function (event, comment) {
+      // TODO: insert efficiently
+      if (comment.parent) {
+        for (var i = 0; i < $scope.comments.length; i++) {
+          if ($scope.comments[i].id == comment.parent) {
+            if (!$scope.comments[i].replies)
+              $scope.comments[i].replies = [];
+            $scope.comments[i].replies.push(comment);
+            break;
+          }
+        }
```

Comment:
Try to see if you can use [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to clean some of this up. In fact, it's worth looking up [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map), [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map), and [forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) as well. They really help to make JS code a lot cleaner and easier to reason with. [Lodash](https://lodash.com/docs#filter) also has these functions but also supports them for objects